src/proof/drat/drat_proof.cpp

```diff
@@ -35,12 +35,20 @@ SatLiteral parse_binary_literal(std::string::const_iterator& start,
 {
   // lit is encoded as uint represented by a variable-length byte sequence
   uint64_t literal_represented_as_uint = 0;
-  for (int shift = 0; start != proof_end; ++start, shift += 7)
+  for (uint8_t shift = 0; start != proof_end; ++start, shift += 7)
   {
+    // Check whether shift is so large that we're going to lose some
+    // information
+    if (shift + 7 > 64)
```

Comment:
This must be actually `if (shift + 7 >= 64)` as it is undefined behavior if it is equal or greater than the bit width (or if it is negative).

Reply:
Good catch!. I'll fix it.